src/parser.rs

```diff
@@ -1854,20 +1847,17 @@ impl Parser {
     }
 
     /// Parse a LIMIT clause
-    pub fn parse_limit(&mut self) -> Result<Option<Expr>, ParserError> {
+    pub fn parse_limit(&mut self) -> Result<Option<u64>, ParserError> {
         if self.parse_keyword("ALL") {
             Ok(None)
         } else {
-            self.parse_literal_uint()
-                .map(|n| Some(Expr::Value(Value::Long(n))))
+            Ok(Some(self.parse_literal_uint()?))
         }
     }
 
     /// Parse an OFFSET clause
-    pub fn parse_offset(&mut self) -> Result<Expr, ParserError> {
-        let value = self
-            .parse_literal_uint()
-            .map(|n| Expr::Value(Value::Long(n)))?;
+    pub fn parse_offset(&mut self) -> Result<u64, ParserError> {
```

Comment:
Same here. This was added in this PR https://github.com/andygrove/sqlparser-rs/pull/69#issuecomment-497160698

Reply:
Backed this bit out too.